src/DistributionsAD.jl

```diff
@@ -49,14 +43,39 @@ export TuringScalMvNormal,
 
 include("common.jl")
 include("univariate.jl")
+#=
 include("multivariate.jl")
 include("mvcategorical.jl")
 include("matrixvariate.jl")
 include("flatten.jl")
 include("arraydist.jl")
 include("filldist.jl")
-@init @require ReverseDiff = "37e2e3b7-166d-5795-8a7a-e32c996b4267" begin
-    include("reversediff.jl")
+=#
+function __init__()
+    @require ForwardDiff="f6369f11-7733-5829-9624-2563aa707210" begin
+        using ForwardDiff: @define_binary_dual_op # Needed for `eval`ing diffrules here
+        include("forwarddiff.jl")
+    end
+
+    @require ReverseDiff = "37e2e3b7-166d-5795-8a7a-e32c996b4267" begin
+        include("reversediff.jl")
+    end
+
+    @require Tracker="9f7883ad-71c0-57eb-9f7f-b5c9e6d3789c" begin
+        using DiffRules
+        using ZygoteRules
+        using SpecialFunctions
+        using LinearAlgebra: AbstractTriangular
+        using Tracker: Tracker, TrackedReal, TrackedVector, TrackedMatrix,
+                       TrackedArray, TrackedVecOrMat, track, @grad, data
+        include("tracker.jl")
+    end
+
+    @require Zygote="e88e6eb3-aa80-5325-afca-941959d7151f" begin
+        using ForwardDiff
+        using Zygote: @adjoint, pullback
+        include("zygote.jl")
+    end
```

Comment:
We should not introduce a Requires block for Zygote (in general, Requires is not great and increases loading times a lot, so it's good if we can avoid it). ZygoteRules contains `@adjoint` and `pullback` (they are just reexported by Zygote), and it's a very lightweight package specifically designed so that packages can write custom adjoints but don't have to depend on Zygote or use Requires (same approach as ChainRulesCore, basically). So we should just load `zygote.jl` outside of the `__init__` function in the regular way.

The only problem is that again some custom adjoints call into a different AD backend (here ForwardDiff) that users have to load to be able to use these adjoints. I'm still not sure what's the best way to solve this issue (I was speculating about some dummy methods that display an informative error message but could be redefined/changed when ForwardDiff is loaded).

In a second separate PR one could try to migrate from ZygoteRules to ChainRulesCore (since Zygote supports ChainRules), but that should not be done in this PR.

Reply:
Yes, I agree, I introduced that block because of the ForwardDiff dependency. The only adjoint that needs it is the one of [`poissont_pdf_fft`](https://github.com/TuringLang/DistributionsAD.jl/blob/809bdc7ba7ee46363cfa8b226e9d5b2598469056/src/univariate.jl#L287-L294). There is some code that is (or maybe was?) no working with Zygote below the part that I linked.

Reply:
I have removed the Zygote require block, as suggested and moved the `@adjoint` for the `poissonbinomial_pdf_fft` to `zygote_forwarddiff.jl` which is in the ForwardDiff require block. Note that all the ForwardDiff code will be loaded together with Zygote, because Zygote depends on ForwardDiff... so the dummy `@adjoint` in `zygote.jl` will practically never be hit.

Reply:
Actually, I think we could just move the zygote adjoints back to the `univaritate.jl` file as it is not loaded conditionally any more

Reply:
I think it's better to keep them in separate files, regardless if they're loaded conditionally or not. 

Regarding `poissonbinomial_pdf_fft`, I guess you could indeed just add a comment to the non-dummy version that explains that Zygote loads ForwardDiff, and hence it will always be available for Zygote. Maybe also add a comment in the Zygote file, that actually one adjoint is defined in the optional ForwardDiff code.